### PR TITLE
Refactor metadata extraction for `ScanImageImagingInterface`

### PR DESCRIPTION
**Pull request by h-mayorquin**
I am trying to add the gin test for `ScanImageImagingInterface` #568 

However, the metadata that we have right for `ScanImageImagingInterface` does not follow the the right format (the problem is that our format expected lists for imaging plane and two photon series and Ben added the metadata as dicts) and therefore the metadata validation fails.

The PR with the added test comes chained to this. 

As a bonus, one of the warnings that was generated by improper use of `dict_deep_update` is removed

### CODE REVIEW

File: src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py [hunk 1]
```diff
@@ -49,12 +50,17 @@ def __init__(
...
+        metadata = super().get_metadata()
 
-        new_metadata = dict(Ophys=dict(TwoPhotonSeries=dict(description=json.dumps(self.image_metadata))))
+        extracted_session_start_time = dateparse(self.image_metadata["state.internal.triggerTimeString"])
+        metadata["NWBFile"] = dict(session_start_time=extracted_session_start_time)
 
-        if "state.internal.triggerTimeString" in self.image_metadata:
-            new_metadata["NWBFile"] = dict(
-                session_start_time=dateparse(self.image_metadata["state.internal.triggerTimeString"])
-            )
```
Comment (bendichter) [inline src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py:58]: why did you change this?

Comment (bendichter) [reply src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py:58]: Now if this field is not present the entire conversion fails

Comment (h-mayorquin) [reply src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py:58]: When I modified the metadata to wrap the dicts into list I was getting problems matching the data to the expected results in the test (as everything was string) so I re-wrote everything to make the comparison specific to each of the levels. The failure mode that you pointed out is something that did not foresee. 

This could be fixed either by testing the present of the extracted_session_time or by writing everything to a temporary metadata dict and then using `deep_dict_update` as you had it before. I prefer the former, do you have a preference?

Comment (h-mayorquin) [reply src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py:58]: Btw, while lookin at the extracted json I saw the following:

```
    "state.internal.startupTimeString": "\'10/9/2017 14:38:30.957\'",
    "state.internal.triggerTimeString": "\'10/9/2017 16:57:07.967\'",
```
There are two time strings over there, how did you chose between those two?

Comment (bendichter) [reply src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py:58]: I would guess the data collection starts at the trigger time

Comment (bendichter) [reply src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py:58]: I don't really have a preference for how we add the data to the dict as long as we are robust to missing metadata

Comment (h-mayorquin) [reply src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py]: Good. Now I added that. I also added a new assertion in case the sampling frequency was not found.

**Comment by CodyCBakerPhD** [comment]
retriggering tests

File: src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py [hunk 2]
```diff
@@ -10,7 +11,16 @@
     HAVE_PIL = False
 
 from ..tiff.tiffdatainterface import TiffImagingInterface
```
Comment (CodyCBakerPhD) [inline src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py:13]: @h-mayorquin Sorry, I'm just now realizing - this is using the `TiffImagingExtractor` instead of the `ScanImageTiffImagingExtractor`; the second one is preferred since it supports memmaping of these files whereas the first one loads the entire file as an array.

Also, as a bonus, the `ScanImageTiffImagingExtractor` has an attribute `_metdata` that has that description string.

Also also, this would swap the dependency from `PIL` to `scanimagetiff` (see ROIExtractors for detail).

Comment (h-mayorquin) [reply src/nwb_conversion_tools/datainterfaces/ophys/scanimage/scanimageimaginginterface.py:13]: Good catch, I did not notice this either.

https://github.com/catalystneuro/nwb-conversion-tools/pull/586

**Comment by CodyCBakerPhD** [comment]
LGTM - @h-mayorquin is this good to go?

**Comment by h-mayorquin** [comment]
> LGTM - @h-mayorquin is this good to go?

Yes.

**Review by CodyCBakerPhD — APPROVED** [review]
